ksql-cli/src/main/java/io/confluent/ksql/cli/console/CommentStripper.java

```diff
@@ -29,37 +29,84 @@ private CommentStripper() {
   }
 
   static String strip(final String line) {
+    return new Parser(line).parse();
+  }
+
+  private static final class Parser {
+    private final String line;
+    private int pos;
+
+    private Parser(final String line) {
+      this.line = line;
+      this.pos = 0;
+    }
+
+    private String parse() {
+      final String firstPart = strip();
+      if (done()) {
```

Comment:
The reason this isn't wrapped into the while-loop below is so we avoid creating a `StringBuilder` instance when the line contains no comments, yes? How expensive is it to create a `StringBuilder` instance? (Trying to improve my understanding of when such optimizations make sense.)

Reply:
Most lines won't have comments, hence I added this early out as its easy to do. 

One could argue this is 'premature optimisation' - especially considering this is, in general, reacting to someone typing at a keyboard and hence performance isn't really an issue.  So asking me to remove this would be a valid point.